### DISCUSSION TITLE
Which keystorage can be used when fips is enabled

### DISCUSSION THREAD

**Discussion by lbermes** [openssl/openssl#21390] [Q&A] [answered]
Hi, I'm stuck with the #19798 because I wanted to load a pkcs12 without the mac check in fips mode with openssl 3.0.9
Is there an alternative to parse the file. An alternative could be to use the key from an excrypted file. I used rsa to create one but it seems that this always ends in evp_pkey_get_legacy which is also not supported in fips mode.
Is there a way to store and load keys in fips mode encrypted?

**Comment by t8m** [comment, marked as answer]
PKCS#8 private key files (in PEM or DER format) can be read in FIPS mode without problems if the default encryption algorithm is used.

**Comment by lbermes** [comment]
Thank you, it works.
